tests/test_vectors.py

```diff
@@ -0,0 +1,170 @@
+import json
+import os
+
+from umbral.curvebn import CurveBN
+from umbral.point import Point, unsafe_hash_to_point
+from umbral.keys import UmbralPublicKey
+from umbral.config import default_params
+from umbral.fragments import KFrag, CapsuleFrag
+from umbral import pre
+
+def test_curvebn_operations():
+
+    vector_file = os.path.join('vectors', 'vectors_curvebn_operations.json')
+    try:
+        with open(vector_file) as f:
+            vector_suite = json.load(f)
+    except OSError:
+        raise 
+
+    bn1 = CurveBN.from_bytes(bytes.fromhex(vector_suite['first operand']))
+    bn2 = CurveBN.from_bytes(bytes.fromhex(vector_suite['second operand']))
+
+    expected = dict()
+    for op_result in vector_suite['vectors']:
+        result = bytes.fromhex(op_result['result'])
+        expected[op_result['operation']] = CurveBN.from_bytes(result)
+
+    test = [  ('Addition', bn1 + bn2),
+              ('Subtraction', bn1 - bn2),
+              ('Multiplication', bn1 * bn2),
+              ('Division', bn1 / bn2), 
+              ('Pow', bn1 ** bn2),
+              ('Mod', bn1 % bn2),
+              ('Inverse', ~bn1),    
+              ('Neg', -bn1),
+            ]
+
+    for (operation, result) in test:
+        assert result == expected[operation], 'Error in {}'.format(operation)
+
+def test_curvebn_hash():
+
+    vector_file = os.path.join('vectors', 'vectors_curvebn_hash.json')
+    try:
+        with open(vector_file) as f:
+            vector_suite = json.load(f)
+    except OSError:
+        raise 
+
+    params = default_params()
+
+    for vector in vector_suite['vectors']:
+        hash_input = [bytes.fromhex(item['bytes']) for item in vector['input']]
```

Comment:
Or do we just feed `bytes` here instead of actual classes to test hashing?

Reply:
My initial intention was to provide a way to test how this hash function works, where the input can be bytes or other crypto objects. To this respect, @karce expressed similar concerns to yours in the associated goUmbral PR, since it would make the test to be more than a unit test, in the sense that it would depend on the correctness of serialization and deserialization of the criptographic elements. Since in the end the hash function transforms everything to bytes, tests can simply input the bytes representation and forget about the specified class, which can be seen just as auxiliary metadata. You can see an example of this working in the tests contained in this PR, which check the generated test vectors.

Reply:
In goUmbral's implementation of these tests it simply ignores the class field since these classes aren't in goUmbral's implementation of Hash and it won't affect the result.